### PR TITLE
[SYCL][libdevice] Keep sign bit of result for std::sin(-0)

### PR DESCRIPTION
**Pull request by jinge90**
For input '-0', __spirv_ocl_sin returns +0 for float and returns -0 for double, MSVC complex math implementation does depend on this, discarding the signbit for sin(-0) will lead to some corner case failures in exp(std::complex<float>) e2e tests. This patch is a workaround for the issue.

### CODE REVIEW

**Comment by jinge90** [comment]
Hi, @KornevNikita 
This PR should fix the pre-ci failure for https://github.com/intel/llvm/pull/16639
Could you try it?
Thanks very much.

**Comment by KornevNikita** [comment]
@jinge90 I've cherry-picked your commit to my PR, waiting for the pre-commit results.

**Review by KseniyaTikhomirova — APPROVED** [review]
LGTM but libdevice should be reviewed by another team, not SYCL RT

**Comment by Naghasan** [comment]
> _spirv_ocl_sin returns +0 for float and returns -0 for double, although no definition exists in standard for -0 input

There is and it should returns -0, the driver has a bug. See https://registry.khronos.org/OpenCL/specs/3.0-unified/html/OpenCL_C.html#edge-case-behavior, and C99 F.9.1.6 states `sin(-0)` returns `-0`.

**Comment by bader** [comment]
> libdevice should be reviewed by another team, not SYCL RT

@KseniyaTikhomirova, I disagree with this statement. `libdevice` code is owned by SYCL RT team for the last 3+ years. If you want to transfer the ownership to another team, please, open a PR changing https://github.com/intel/llvm/blob/sycl/.github/CODEOWNERS. The PR must be approved by both teams SYCL RT and new `libdevice` code owner team.

**Comment by jinge90** [comment]
> > _spirv_ocl_sin returns +0 for float and returns -0 for double, although no definition exists in standard for -0 input
> 
> There is and it should return -0, the driver has a bug. See https://registry.khronos.org/OpenCL/specs/3.0-unified/html/OpenCL_C.html#edge-case-behavior, and C99 F.9.1.6 states `sin(-0)` returns `-0`.

Hi, @Naghasan 
Thanks very much for pointing out this! I will report this to GPU driver developers and before the driver bug is fixed, we will use this workaround to fix complex math's corner case failure.

File: libdevice/fallback-cmath.cpp [hunk 1]
```diff
@@ -168,8 +168,13 @@ float __devicelib_fmaf(float x, float y, float z) {
   return __spirv_ocl_fma(x, y, z);
 }
 
+#if !defined(__NVPTX__) && !defined(__AMDGCN__)
```
Comment (Maetveis) [inline libdevice/fallback-cmath.cpp:171]: The condition shouldn't be inverted. For future extension any potential new target should not be assumed to be buggy.

```suggestion
#if defined(__SPIR__) || defined(__SPIRV__)
```

**Review by Maetveis — CHANGES_REQUESTED** [review]

File: libdevice/fallback-cmath.cpp (hunk 1 above)
Comment (jinge90) [reply libdevice/fallback-cmath.cpp]: Hi, @Maetveis 
Fixed.
Thanks very much.

**Review by Maetveis — APPROVED** [review]

**Comment by Maetveis** [comment]
LGTM, thanks!

**Comment by jinge90** [comment]
Hi, @intel/llvm-gatekeepers 
Could you help merge this PR?
Thanks very much.
